src/main/java/org/stellar/sdk/Server.java

```diff
@@ -264,6 +272,65 @@ public SubmitTransactionResponse submitTransaction(Transaction transaction) thro
         return submitTransactionResponse;
     }
 
+    /**
+     * Submits transaction to the network
+     *
+     * This function will always check if the destination account requires a memo in the transaction as
+     * defined in <a href="https://github.com/stellar/stellar-protocol/blob/master/ecosystem/sep-0029.md" target="_blank">SEP-0029</a>
+     * If you want to skip this check, use {@link Server#submitTransaction(Transaction, boolean)}.
+     *
+     * @param transaction transaction to submit to the network.
+     * @return {@link SubmitTransactionResponse}
+     * @throws SubmitTransactionTimeoutResponseException When Horizon returns a <code>Timeout</code> or connection timeout occured.
+     * @throws SubmitTransactionUnknownResponseException When unknown Horizon response is returned.
+     * @throws AccountRequiresMemoException when a transaction is trying to submit an operation to an
+     * account which requires a memo.
+     * @throws IOException
+     */
+    public SubmitTransactionResponse submitTransaction(Transaction transaction) throws IOException, AccountRequiresMemoException {
+        return submitTransaction(transaction, false);
+    }
+
+    /**
+     * checkMemoRequired implements a memo required check as defined in
+     * <a href="https://github.com/stellar/stellar-protocol/blob/master/ecosystem/sep-0029.md" target="_blank">SEP-0029</a>
+     *
+     * @param transaction transaction to submit to the network.
+     * @throws AccountRequiresMemoException when a transaction is trying to submit an operation to an
+     * account which requires a memo.
+     * @throws IOException
+     */
+    public void checkMemoRequired(Transaction transaction) throws IOException, AccountRequiresMemoException {
+        if (!(transaction.getMemo() == null || transaction.getMemo().equals(Memo.none()))) {
+            return;
+        }
+        Set<String> destinations = new HashSet<String>();
+        Operation[] operations = transaction.getOperations();
+        for (int i = 0; i < operations.length; i++) {
+            String destination;
+            Operation operation = operations[i];
+            if (operation instanceof PaymentOperation) {
```

Comment:
Is there a better way to do this under Java 6?

Reply:
I don't think so